packages/dds/tree/src/simple-tree/ProxyBinding.md

```diff
@@ -0,0 +1,99 @@
+# simple-tree Proxy Binding
+
+Proxy binding is a technique used in the simple-tree API implementation which enables a useful API feature, namely the ability to use a node proxy after it is inserted into the tree.
+This allows developers to "read back" values that they insert into the tree much more succinctly.
+Here is an example:
+
+```ts
+function addPoint(curve: Curve, x: number, y: number): Point {
+	const point = new Point({ x: 3, y: 3 });
+	curve.points.insertAtEnd(point);
+	// After insertion, `point` can be queried:
+	assert(point.x === 3);
+	// In fact, `point` is the same proxy object that you would get from reading it off of its new parent in the tree:
+	assert(point === curve.points[curve.points.length - 1]);
+	// So, to read the content that was just inserted, the original object can be used and there is no need to read via the parent:
+	return point;
+	// (rather than: `return curve.points[curve.points.length - 1]`)
+}
+```
+
+## Implementation
+
+This feature is supported by doing some bookkeeping to ensure that the proxy objects,
+flex nodes and anchor nodes in the tree get associated and disassociated at the right times.
+There are three states that a node proxy can be in: "raw", "marinated" and "cooked".
+
+### Raw Proxies
+
+A newly created proxy, a.k.a. a **raw** proxy. A raw proxy is produced by invoking the schema-provided constructor for a node:
+
+```ts
+const rawPoint = new Point({ x: 3, y: 3 });
+```
+
+Such a proxy will be raw until it is inserted into the tree and becomes "marinated" (see below).
+As of this writing raw proxies cannot be read or mutated, but in the future they will be capable of both.
```

Comment:
Great idea about the timestamp, I will update.

And yes, currently if you do something like:

`const foo = new Foo({ name: "foo" })`

and then try to do `foo.name`, it will throw. It's pretty lame, so I'm hoping to implement reading for those soon.